File: leetcode/questions/75/python/75.py
class Solution:
    def sortColors(self, nums):
        len_nums = len(nums)
        idx, color_a, color_b = 0, 0, len_nums - 1

        while idx <= color_b:
            if nums[idx] == 0:
                nums[idx], nums[color_a] = nums[color_a], nums[idx]
                color_a += 1
                idx += 1

            elif nums[idx] == 2:
                nums[idx], nums[color_b] = nums[color_b], nums[idx]
                color_b -= 1

            else:
                idx += 1

File: leetcode/questions/75/java/75.java
class Solution {
    private void swap(int[] arr, int idxA, int idxB) {
        int tmp = arr[idxA];
        arr[idxA] = arr[idxB];
        arr[idxB] = tmp;
    }

    public void sortColors(int[] nums) {
        int numsLen = nums.length;
        int colorA = 0, colorB = numsLen - 1, idx = 0;

        while (idx <= colorB) {
            if (nums[idx] == 0) {
                swap(nums, idx++, colorA++);
            } else if (nums[idx] == 2) {
                swap(nums, idx, colorB--);
            } else {
                ++idx;
            }
        }
    }
}